backend/app/utils/ig_related_services.py

```diff
@@ -0,0 +1,94 @@
+"""Helper module to inject related services data into interoperability guileliens response"""
+import asyncio
+import logging
+from typing import Optional
+
+from httpx import AsyncClient, ConnectError, ConnectTimeout
+
+from app.consts import ResourceType
+from app.error_handling.exceptions import RelatedServicesError
+from app.schemas.solr_response import Collection, RelatedService
+from app.settings import settings
+from app.solr.operations import get_item_by_pid
+
+logger = logging.getLogger(__name__)
+
+
+def _parse_categories(categories: list, unified_categories: Optional[list]) -> list:
+    categories_set = set()
+    for item in categories:
+        try:
+            categories_set.add(item.split(">")[1])
+        except IndexError:
+            pass
+    if unified_categories:
+        return list(categories_set.union(unified_categories))
+    return list(categories_set)
+
+
+async def _get_related_records_pids(client, ig_pid):
+    try:
+        response = await client.get(
+            f"{settings.RELATED_SERVICES_ENDPOINT}/{ig_pid}",
+        )
+        if response.status_code != 200:
+            logger.error("%s. %s", response.status_code, str(response))
+            raise RelatedServicesError(
+                status_code=response.status_code,
+                detail=(
+                    f"Related services for guideline {ig_pid} server status error: \n\n"
+                    f" {response}"
+                ),
+            )
+        return response.json()
+
+    except (ConnectError, ConnectTimeout) as e:
+        logger.error("%s", str(e))
+        raise RelatedServicesError(detail="Connection error") from e
+
+
+async def extend_ig_with_related_services(client: AsyncClient, docs: list[dict]):
+    """Main function responsible for extending iteroperability guideline response
+    with related services data
+    """
+    for doc in docs:
+        if doc["type"] == ResourceType.GUIDELINE:
+            related_services_pids = await _get_related_records_pids(client, doc["id"])
```

Comment:
we don't want to catch the `RelatedServicesError` somewhere?

Reply:
and returning an empty array as a default behaviour on catch (after logging the error)